GW2EIEvtcParser/EncounterLogic/Fractals/SunquaPeak/AiKeeperOfThePeak.cs

```diff
@@ -238,6 +239,27 @@ internal override void EIEvtcParse(ulong gw2Build, EvtcVersionEvent evtcVersion,
         }
     }
 
+    internal override long GetFightOffset(EvtcVersionEvent evtcVersion, FightData fightData, AgentData agentData, List<CombatItem> combatData)
+    {
+        // check invulnerability remove for new elemental ai
+        var ai = agentData.GetNPCsByID(TargetID.AiKeeperOfThePeak).FirstOrDefault() ?? throw new MissingKeyActorsException("Ai not found");
+        var invulnStart = GetFightOffsetByInvulnStart(fightData, combatData, ai, Determined895);
+        if (invulnStart > ai.FirstAware)
+        {
+            return invulnStart;
+        }
+
+        // first cast or fallback to regular offset (combat enter) for dark ai
+        // old elemental ai will always end up with fallback due to idle time
+        var start = base.GetFightOffset(evtcVersion, fightData, agentData, combatData);
+        var firstCast = combatData.Where(x => x.IsActivation != Activation.None && x.SrcMatchesAgent(ai)).FirstOrDefault();
```

Comment:
Should also verify IsStateChange == None, no guarantee that other state changes won't use IsActivation byte for something..